aurora_module/rds_aurora_cluster.tf

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2011 - 2019, Coveo Solutions Inc.
+ */
+
+locals {
+  instances_count = "${var.engine_mode == "serverless" ? 0 : var.instances_count}"
+}
+
+resource "aws_db_subnet_group" "db_subnet_group" {
+  subnet_ids = "${var.subnet_ids}"
+
+  name        = "${lookup(var.optional_parameters, "db_subnet_group_name", "${var.custom_identifier}")}"
+  description = "${lookup(var.optional_parameters, "subnet_group_description", "${var.custom_identifier}")}"
+  tags        = "${var.subnet_group_tags}"
+}
+
+resource "aws_kms_key" "rds_cluster_encryption_key" {
+  tags = "${var.db_tags}"
+}
+
+resource "aws_kms_alias" "rds_cluster_encryption_key_alias" {
+  target_key_id = "${aws_kms_key.rds_cluster_encryption_key.key_id}"
+  name          = "alias/${lookup(var.optional_parameters, "rds_cluster_encryption_key_alias_name", "${var.custom_identifier}_dbek")}"
+}
+
+resource "aws_kms_key" "rds_cluster_master_passw_encryption_key" {
+  tags = "${var.db_tags}"
+}
+
+resource "aws_kms_alias" "rds_cluster_master_passw_encryption_key_alias" {
+  target_key_id = "${aws_kms_key.rds_cluster_master_passw_encryption_key.key_id}"
+  name          = "alias/${lookup(var.optional_parameters, "rds_cluster_master_passw_encryption_key_alias_name", "${var.custom_identifier}_mpek")}"
+}
+
+resource "aws_ssm_parameter" "rds_cluster_master_username" {
```

Comment:
What if I want to use the AWS Secrets Manager and a Lambda to handle my user password rotation? 🙂 

IMO, storing the credentials in SSM should be an option.

Reply:
Good point, I didn't think about the Secret Manager service. I'll revisit that.

Reply:
@JPLachance I updated my PR with secret manager. Let me know what you think. I've tested it and it works well.

Reply:
I think there was a miss-understanding. From my point of view, this Terraform module should create the database and that's it. The user of the module should do what he wants with the database credentials outside of this module. 😄 

That being said, this is now configurable and not mandatory, so ship it!

Reply:
Maybe I should rename the module to something more explicit.
My goal here is to limit the number of resources we have to create when we start a project. One module to provide as must as required resources as possible to prevent re-create everything for a new service. 😄